tiledb/sm/c_api/tiledb.h

```diff
@@ -1097,6 +1097,13 @@ TILEDB_EXPORT int32_t tiledb_query_set_subarray_t(
  * either hold the values to be written (if it is a write query), or will hold
  * the results from a read query.
  *
+ * The caller owns the `buffer` provided and is responsible for freeing the
+ * memory associated with it. For writes, the buffer holds values to be written
+ * which can be freed at any time after query completion. For reads, the buffer
+ * is allocated by the client and will contain data read by the query after
+ * completion. The freeing of this memory is up to the user once they are done
+ * referencing the read data.
```

Comment:
Can you use a single word for "caller", "user" and "user"?

Reply:
Done, thanks!